test_autofit/graphical/functionality/test_messages.py

```diff
@@ -98,4 +142,107 @@ def test_meanfield_gradients():
         norm = np.linalg.norm(grad[v] - njac1[v].sum((0, 1)))
         assert norm == pytest.approx(0, abs=1e-2)
         norm = np.linalg.norm(grad[v] - njac2[v].sum(0))
-        assert norm == pytest.approx(0, abs=1e-2)
\ No newline at end of file
+        assert norm == pytest.approx(0, abs=1e-2)
+
+
+def test_beta():
+    a = b = np.r_[2., 3.2, 1.5]
+    beta = graph.BetaMessage(a, b[::-1])
+    check_dist_norms(beta)
+
+    betas = [
+        graph.BetaMessage(a, b)
+        for a, b in (np.random.poisson(5, size=(10, 2)) + 1)
+    ]
+    [check_dist_norm(b) for b in betas]
+    check_log_normalisation(betas)
+
+
+def test_message_transforms():
+    a = b = np.r_[2., 3.2, 1.5]
+    mx, sigma = np.r_[-0.5, .1, 0.], np.r_[0.3, 0.5, 0.9]
+    test_cases = [
+        (graph.BetaMessage, (a, b)),
+        (graph.NormalMessage, (mx, sigma)),
+        (graph.UniformNormalMessage, (mx, sigma)),
+    ]
+    shift = (0.3, 5.1)
+    beta = graph.BetaMessage.shifted(0.3, 5.1)(a, b[::-1])
+
+
+def check_transforms(transform, x):
+    y, logd, logd_grad, jac = transform.transform_det_jac(x)
+
+    njac = numerical_jacobian(x, transform.transform)
+    nlogd_grad = numerical_jacobian(x, transform.log_det)
+
+    assert np.allclose(transform.inv_transform(y), x)
+    assert np.allclose(jac.to_dense(), njac)
+    assert np.isclose(logd.sum(), np.linalg.slogdet(njac)[1])
+    assert np.allclose(nlogd_grad.sum(0), logd_grad)
+
+
+def test_transforms():
+    tests = [
+        (transform.log_transform, np.r_[10, 2, 0.1]),
```

Comment:
Might make sense to use pytest.mark.parametrize for this

Reply:
potentially, though pytest.mark.parametrize doesn't work well with my personal Jupyter Notebook workflow

Reply:
I also don't know how to use `pytest.mark.parametrize` ...

Reply:
```python
@pytest.mark.paramtrize(
    "arg1, arg2",
   [('value a 1', 'value a 2'), ('value b 1', 'value b 2')]
):
def test(arg1, arg2):
    ...
```

It generates a test for each arg tuple in the list